cloud/services/scalesets/vmss.go

```diff
@@ -77,6 +77,20 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 		return errors.New("invalid VMSS specification")
 	}
 
+	vmss, err := s.Client.Get(ctx, vmssSpec.ResourceGroup, vmssSpec.Name)
```

Comment:
Probably you can do 

_, err = <> 

since the vmss output isnt used anywhere, just to avoid confusion